Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchJob.java

```diff
@@ -1040,6 +919,28 @@ public boolean isActive() throws IOException {
                 .map(yarnApp -> yarnApp.getLogAggregationStatus().toUpperCase());
     }
 
+    public Observable<Integer> getYarnContainerLogUrlPort() {
+        final int DEFAULT_YARN_CONTAINER_LOG_URL_PORT = 30060;
+        return getSparkJobApplicationIdObservable()
```

Comment:
Could you implement this as a lazy evaluation?

Reply:
`getYarnContainerLogUrlPort` is only called once so lazy evaluation is not necessary.